### DISCUSSION TITLE
Challenge 4 - Decimal Rounding

### DISCUSSION THREAD

**Discussion by surajgjadhav** [Cyfrin/foundry-full-course-cu#95] [Q&A] [answered]
In the Challenge 4, Contract price decimal are rounded to 19 (by multiplying price with 10e10) decimals instead of 18.
As per my limited knowledge 😅 , it should be 18 decimals so that we can use +- 3e18 range (to keep price feed in range of 18 decimals) and check price.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31582237/bf44691c-67f2-470c-8e7d-913f893154d7)

what's your thought on this @PatrickAlphaC ?
Thanks!

**Comment by cromewar** [comment]
Hello @surajgjadhav
As you put on the `getDecimals()` the number of decimals is 8, and the get `getPrice()` give you an output like this:
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/20199180/d73a3265-b9e3-4843-8b8f-ce04d77a825d)
If you notice, the final count is 18 decimals.

Am I missing something here @PatrickAlphaC ?

**Reply by surajgjadhav** [reply]
Hi @cromewar ,
There might be confusion about the Contract.
I am talking about this Challenge 4 Contract: https://sepolia.etherscan.io/address/0x1b30DA2a868704483143A4D46865Ac9585629fd0#code

**Reply by PatrickAlphaC** [reply, marked as answer]
I try not to give too many hints on the challenge contracts ;) 

However,

10e10 * 10e8 == 10e18 not 10e19

**Reply by surajgjadhav** [reply]
Oh! That's tricky.
Thank You so much @PatrickAlphaC !! 😄

**Reply by utkuerkin** [reply]
But how? 10e10 * 10e8 == 10e18 returns false in the Javascript console. We can rewrite the equation as 10*10*e10*e8 which is 100*e18 which is 10*e19. What am I missing here?

**Reply by surajgjadhav** [reply]
HI @PatrickAlphaC ,
I tried that multiplication in remix and I got 1e19.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/31582237/efb7186b-b87b-476e-bf01-050761a36b36)

**Reply by RichardPalladino** [reply]
it's not 1e8 * 10e10 == 10e18?

**Comment by RichardPalladino** [comment]
> I try not to give too many hints on the challenge contracts ;)
> 
> However,
> 
> 10e10 * 10e8 == 10e18 not 10e19

Isn't it 1e8 * 10e10 == 10e18?
